### PR TITLE
lock: add delay clean table lock when session close

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
Add `delay-clean-table-lock` configuration to control delay clean table lock.

It is used for `TiSpark`.

Before This PR: 

The logic of `TiSpark` to use the table lock is:

1. lock table
2. prewrite data.
3. check whether the locked table session is closed.
4. if the session doesn't close, do commit. 
5. if the session was closed by unexpected, do a rollback.

The problem is, What if the session was closed by unexpected between step 3 and 4, It will cause some problem. 

TiFlash need to delayed release table lock before do commit.
 

### What is changed and how it works?

Add `sleep` in `session.Close` if `delay-clean-table-lock` has been configed.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch 3.0

### CODE REVIEW

File: config/config.toml.example [hunk 1]
```diff
@@ -57,6 +57,9 @@ treat-old-version-utf8-as-utf8mb4 = true
 # enable-table-lock is used to control table lock feature. Default is false, indicate the table lock feature is disabled.
 enable-table-lock = false
 
+# delay-clean-table-lock is used to control delay release table lock when the abnormal situation. (Milliseconds)
```
Comment (winkyao) [inline config/config.toml.example]: `is used to control delay release table lock when the abnormal situation.` -> `is used to control whether delayed-release the table lock in the abnormal situation.`

Comment (crazycs520) [inline config/config.toml.example]:
```suggestion
# delay-clean-table-lock is used to control whether delayed-release the table lock in the abnormal situation. (Milliseconds)
```

**Pull request merged**